roles/apt/tasks/main.yml

```diff
@@ -28,21 +27,8 @@
   with_fileglob:
   - files/apt.conf.d/*
 
-# d-i puts the apt config in apt.conf, but we prefer a .d file
-- name: ensure apt proxy isn't in apt.conf
-  lineinfile:
-    dest: /etc/apt/apt.conf
-    regexp: ^Acquire::http::Proxy
-    state: absent
-
-- name: setup apt proxy
-  template:
-    src: templates/30apt-proxy.j2
-    dest: /etc/apt/apt.conf.d/30apt-proxy
-  when: apt_proxy != ""
-
 - name: unconfigure apt proxy
   file:
     path: /etc/apt/apt.conf.d/30apt-proxy
     state: absent
-  when: apt_proxy == ""
+  when: not apt_proxy
```

Comment:
I assume you can delete this whole block, because nothing is ever going to create that file any more.